src/js/components/projects/projectForm.tsx

```diff
@@ -26,11 +28,13 @@ const ProjectForm = ({ productName, doPostObject }: Props) => {
       setHasName(false);
       return;
     }
-    // check if project (name) exists?
     doPostObject({
       objectType: OBJECT_TYPES.PROJECT,
       content: { name, description, commit_message: '', release_notes: '' },
-    });
+    }).finally(() => (
+      // display errors i.e. if project (name) exists?
+      <Redirect to={routes.project_detail(productSlug, productName)} /> // @todo, this isn't redirecting...
+    ));
```

Comment:
@jgerigmeyer a couple things about these couple lines:  
1. Any idea why this redirect is not working? 
2.  The api returns error msgs that are stored in redux.  Would using a selector be the approach for displaying those errors? (i.e. if the `name` already exists for a project).
3.  Would we need need to anticipate long descriptions or project names? I don't see a char limit on those. 
4.  Are there restricted chars for project names?

Reply:
1. `<Redirect>` is a _declarative_ way to handle routing redirects, but that will only work within a `render` block (that is, rendered markup). Instead here you'll want to use `withRouter()` and then `history.push()` to navigate. See e.g. https://github.com/SFDO-Tooling/MetaDeploy/blob/master/src/js/components/plans/ctaButton.js#L215-L228

Related, I think we only want to do that redirect on success (`.then()`), not `.finally()`.

2. Hm, yeah. I don't think we want to store form-specific error messages globally, no. My hunch is that we want to do two things: 1) Make sure the API is returning helpful human-readable error messages in the `details` or `non_field_errors` keys, and 2) Expose field-specific error messages to the component, storing them to display in local component state (not the global redux store). That would also allow us to get rid of the custom `hasName` logic so we can just rely on server-side validation.

I'd be happy to tackle this problem, if you'd like. Or we can talk more about what I mean.

3. Yeah, good questions. @wlonk What do you think about restrictions on length or specific chars?
4. ☝️

Reply:
Some further clarification on points 3-4: Project names are already limited to 50 chars. Descriptions might get long, but that's okay -- we can truncate them for display purposes where we need to (e.g. the list view). And we don't need to worry about restrictions -- we use the `name` to create a `slug` (which is already a munged version), and we can use that for the GH branch name.